src/app/qgisapp.cpp

```diff
@@ -5470,6 +5471,15 @@ QgsMeshLayer *QgisApp::addMeshLayerPrivate( const QString &url, const QString &b
     return nullptr;
   }
 
+  // Manage default reference time, if not reference time is present by default in the layer -> assign one
+  if ( ! layer->temporalProperties()->referenceTime().isValid() )
```

Comment:
Looks good for now! If there's more need for this in future we could consider adding a method to the layer temporal properties, something like "setDefaultsFromProject( QgsProject* project )" which we call here instead, but for now this is fine.